game/missiongenerator/tgogenerator.py

```diff
@@ -97,37 +93,110 @@ def __init__(
     def culled(self) -> bool:
         return self.game.position_culled(self.ground_object.position)
 
-    def generate(self) -> None:
+    def generate(self, unique_name: bool = True) -> None:
         if self.culled:
             return
 
         for group in self.ground_object.groups:
             if not group.units:
                 logging.warning(f"Found empty group in {self.ground_object}")
                 continue
-
-            unit_type = vehicle_type_from_name(group.units[0].type)
-            vg = self.m.vehicle_group(
-                self.country,
-                group.name,
-                unit_type,
-                position=group.position,
-                heading=group.units[0].heading,
-            )
-            vg.units[0].name = group.units[0].name
-            vg.units[0].player_can_drive = True
-            for i, u in enumerate(group.units):
-                if i > 0:
-                    vehicle = Vehicle(self.m.next_unit_id(), u.name, u.type)
-                    vehicle.position.x = u.position.x
-                    vehicle.position.y = u.position.y
-                    vehicle.heading = u.heading
-                    vehicle.player_can_drive = True
-                    vg.add_unit(vehicle)
-
-            self.enable_eplrs(vg, unit_type)
-            self.set_alarm_state(vg)
-            self._register_unit_group(group, vg)
+            group_name = group.group_name if unique_name else group.name
+            if group.static_group:
+                # Static Group
+                for i, u in enumerate(group.units):
+                    if isinstance(u, SceneryGroundUnit):
+                        # Special handling for scenery objects:
+                        # Only create a trigger zone and no "real" dcs unit
+                        self.add_trigger_zone_for_scenery(u)
+                        continue
+
+                    # Only skip dead units after trigger zone for scenery created!
+                    if not u.alive:
+                        continue
+
+                    unit_type = unit_type_from_name(u.type)
+                    if not unit_type:
+                        raise RuntimeError(
+                            f"Unit type {u.type} is not a valid dcs unit type"
+                        )
+
+                    sg = self.m.static_group(
+                        country=self.country,
+                        name=u.unit_name if unique_name else u.name,
+                        _type=unit_type,
+                        position=u.position,
+                        heading=u.position.heading.degrees,
+                        dead=not u.alive,
+                    )
+                    self._register_ground_unit(u, sg.units[0])
```

Comment:
can be merged with the group creation for moving_group. just check there if it is subclass StaticType